### PR TITLE
chore(migration): migrate from CRA to VITE

### PR DESCRIPTION
**Pull request by oyo**
## Description

Replace the deprecated react-scripts with vite framework

## Why

react-scripts is deprecated and we can't update any more. We replace it with the Vite Framework.

## Issue

#537, #506

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [ ] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [ ] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,84 +28,83 @@
...
+    "axios": "^1.6.8",
     "buffer": "^6.0.3",
-    "dayjs": "^1.11.8",
-    "i18next": "^22.5.1",
-    "i18next-browser-languagedetector": "^7.0.2",
-    "keycloak-js": "^23.0.7",
+    "dayjs": "^1.11.10",
+    "i18next": "^23.10.1",
+    "i18next-browser-languagedetector": "^7.2.0",
+    "keycloak-js": "^24.0.1",
```
Comment (evegufy) [inline package.json:43]: Please change back to the 23.0.7 version to be aligned with CX IAM version for the 24.05 release

**Review by evegufy — CHANGES_REQUESTED** [review]
Please update the changelog with a precise description of this change as it has quite some impact

**Comment by evegufy** [comment]
A review is required for npm/npmjs/-/eslint-config-love/43.1.0.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13906 .
A review is required for npm/npmjs/-/es-iterator-helpers/1.0.18.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13907 .
A review is required for npm/npmjs/-/immer/10.0.4.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13908 .
A review is required for npm/npmjs/-/react-hook-form/7.51.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13909 .
A review is required for npm/npmjs/-/reflect.getprototypeof/1.0.6.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13910 .
A review is required for npm/npmjs/-/source-map-js/1.2.0.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13911 .
A review is required for npm/npmjs/-/tsconfck/3.0.3.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13912 .
A review is required for npm/npmjs/-/react-redux/9.1.0.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13913 .
A review is required for npm/npmjs/-/react-player/2.15.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13914 .
A review is required for npm/npmjs/@babel/compat-data/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13915 .
A review is required for npm/npmjs/@babel/helper-module-imports/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13916 .
A review is required for npm/npmjs/@babel/code-frame/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13917 .
A review is required for npm/npmjs/@babel/core/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13918 .
A review is required for npm/npmjs/@babel/generator/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13919 .
A review is required for npm/npmjs/@babel/plugin-syntax-jsx/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13920 .
A review is required for npm/npmjs/@babel/highlight/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13921 .
A review is required for npm/npmjs/@babel/helpers/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13922 .
A review is required for npm/npmjs/@babel/helper-string-parser/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13923 .
A review is required for npm/npmjs/@babel/plugin-syntax-typescript/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13924 .
A review is required for npm/npmjs/@babel/plugin-transform-react-jsx-source/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13925 .
A review is required for npm/npmjs/@babel/traverse/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13926 .
A review is required for npm/npmjs/@mui/utils/5.15.14.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13927 .
A review is required for npm/npmjs/@babel/plugin-transform-react-jsx-self/7.24.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13928 .
A review is required for npm/npmjs/@mui/x-date-pickers/6.19.7.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13929 .
A review is required for npm/npmjs/@types/lodash.uniq/4.5.9.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13930 .
A review is required for npm/npmjs/@types/qs/6.9.13.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13931 .
A review is required for npm/npmjs/@mui/x-data-grid/6.19.6.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13932 .
A review is required for npm/npmjs/@typescript-eslint/eslint-plugin/7.3.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13933 .
A review is required for npm/npmjs/@typescript-eslint/types/7.3.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13934 .
A review is required for npm/npmjs/@typescript-eslint/typescript-estree/7.3.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13935 .
A review is required for npm/npmjs/@typescript-eslint/type-utils/7.3.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13936 .
A review is required for npm/npmjs/@typescript-eslint/scope-manager/7.3.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13937 .
A review is required for npm/npmjs/@typescript-eslint/parser/7.3.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13938 .
A review is required for npm/npmjs/@typescript-eslint/visitor-keys/7.3.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13939 .
A review is required for npm/npmjs/@typescript-eslint/utils/7.3.1.
A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13940 .

**Comment by oyo** [comment]
> Please update the changelog with a precise description of this change as it has quite some impact

done

**Comment by evegufy** [comment]
The following IP issues ran into a manual review by the EF IP team:
npm/npmjs/@mui/utils/5.15.14
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13927
npm/npmjs/@mui/x-data-grid/6.19.6
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13932
npm/npmjs/@mui/x-date-pickers/6.19.7
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13929
npm/npmjs/@types/lodash.uniq/4.5.9
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13930
npm/npmjs/@types/qs/6.9.13
https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/13931

The review will take a couple days. As those dependencies will most certainly be approved and to be able to test the change on dev env, the PR is merged to the dev branch with those "restricted" dependencies to the dev branch. We'll keep monitoring the IP issues and no release will occur with restricted in the DEPENDENCIES file.

**Review by evegufy — APPROVED** [review]

**Pull request merged**